daemon/delete_test.go

```diff
@@ -23,68 +23,61 @@ func newDaemonWithTmpRoot(t *testing.T) (*Daemon, func()) {
-// to remove a container that is restarting (#30842)
-func TestContainerDeleteRestarting(t *testing.T) {
-	c := &container.Container{
-		CommonContainer: container.CommonContainer{
-			ID:     "test",
-			State:  container.NewState(),
-			Config: &containertypes.Config{},
-		},
+// TestContainerDelete tests that a useful error message and instructions is
+// given when attempting to remove a container (#30842)
+func TestContainerDelete(t *testing.T) {
+	tt := []struct {
+		errMsg        string
+		fixMsg        string
+		initContainer func() *container.Container
+	}{
+		// a paused container
+		{
+			errMsg: "cannot remove a paused container",
+			fixMsg: "Unpause and then stop the container before attempting removal or force remove",
+			initContainer: func() *container.Container {
+				return newContainerWithIDAndConfig(&container.State{Paused: true, Running: true})
+			}},
+		// a restarting container
+		{
+			errMsg: "cannot remove a restarting container",
+			fixMsg: "Stop the container before attempting removal or force remove",
+			initContainer: func() *container.Container {
+				c := newContainerWithIDAndConfig(container.NewState())
+				c.SetRunning(0, true)
+				c.SetRestarting(&container.ExitStatus{})
+				return c
+			}},
+		// a running container
+		{
+			errMsg: "cannot remove a running container",
+			fixMsg: "Stop the container before attempting removal or force remove",
+			initContainer: func() *container.Container {
+				return newContainerWithIDAndConfig(&container.State{Running: true})
+			}},
 	}
 
-	c.SetRunning(0, true)
-	c.SetRestarting(&container.ExitStatus{})
-
-	d, cleanup := newDaemonWithTmpRoot(t)
-	defer cleanup()
-	d.containers.Add(c.ID, c)
-
-	err := d.ContainerRm(c.ID, &types.ContainerRmConfig{ForceRemove: false})
-	assert.Error(t, err, "cannot remove a restarting container")
-	assert.Error(t, err, "Stop the container before attempting removal or force remove")
-}
+	for _, te := range tt {
+		c := te.initContainer()
+		d, cleanup := newDaemonWithTmpRoot(t)
+		defer cleanup()
+		d.containers.Add(c.ID, c)
 
-// TestContainerDeleteRunning tests that a useful error message and instructions is given when attempting
-// to remove a running container (#30842)
-func TestContainerDeleteRunning(t *testing.T) {
-	c := &container.Container{
-		CommonContainer: container.CommonContainer{
-			ID:     "test",
-			State:  &container.State{Running: true},
-			Config: &containertypes.Config{},
-		},
+		err := d.ContainerRm(c.ID, &types.ContainerRmConfig{ForceRemove: false})
+		assert.Error(t, err, te.errMsg)
+		assert.Error(t, err, te.fixMsg)
 	}
-
-	d, cleanup := newDaemonWithTmpRoot(t)
-	defer cleanup()
-	d.containers.Add(c.ID, c)
-
-	err := d.ContainerRm(c.ID, &types.ContainerRmConfig{ForceRemove: false})
-	assert.Error(t, err, "cannot remove a running container")
-	assert.Error(t, err, "Stop the container before attempting removal or force remove")
 }
 
 func TestContainerDoubleDelete(t *testing.T) {
```

Comment:
I think this test can also be folded into `TestContainerDelete`. If not, it can at least use `newContainerWithIDAndConfig`.